server/cluster/cluster.go

```diff
@@ -1651,16 +1654,35 @@ func (c *RaftCluster) RemoveStoreLimit(storeID uint64) {
 	}
 	delete(cfg.StoreLimit, storeID)
 	c.opt.SetScheduleConfig(cfg)
+	if err := c.opt.Persist(c.storage); err != nil {
+		log.Error("persist store limit meet error", errs.ZapError(err))
+	}
 }
 
 // SetStoreLimit sets a store limit for a given type and rate.
 func (c *RaftCluster) SetStoreLimit(storeID uint64, typ storelimit.Type, ratePerMin float64) {
+	old := c.opt.GetScheduleConfig().Clone()
 	c.opt.SetStoreLimit(storeID, typ, ratePerMin)
+	if err := c.opt.Persist(c.storage); err != nil {
+		// roll back the store limit
+		c.opt.SetScheduleConfig(old)
```

Comment:
What will happen if the rollback operation fails also?

Reply:
The rollback operation only resets the memory value